### DISCUSSION TITLE
Escline definition

### DISCUSSION THREAD

**Discussion by larsgw** [kdl-org/kdl#20] [General]
In my mind backslashes escape the literal line feed byte so it's a bit weird to see whitespace and comments after the backslash. It's your document language though so if it's a must then it's fine.

**Comment by zkat** [comment]
I'd love to hear more folks' thoughts on this. I agree with you to some extent, but also it seems handy to be able to comment at all EOL places across the language. Otherwise, this would be the only exception.

**Reply by zkat** [reply]
p.s. I know it's "my" document language but also I always really appreciate your feedback (and your help!) and it's always super welcome. I'm looking forward to having a discussion about this. :)

**Reply by larsgw** [reply]
This isn't really an argument either way, but from what I understood the two inspirations of this language are shell commands and XML and neither support comments within their equivalent of nodes. In the end, "it doesn't look right" isn't on the list of priorities though.

**Comment by lilyball** [comment]
I've always wanted the ability to put comments after escaped newlines. If KDL supported escaping stuff in general then this would be a problem, but since the backslash only serves to escape a newline, then I think it's fine to support trailing whitespace and comments.

That said, the current spec is

```
escline := '\\' ws* (single-line-comment | newline)
```

and it feels mildly surprising that I can write

```
foo \ // comment here
  "bar"
```

but I can't write (Edit: yes I can, see comments below)

```
foo \ /* comment here */
  "bar"
```

However, supporting the latter then raises the issue of how to handle something like

```foo
this \ /*
  comment
  */ "bar"

or-this \ /*
  comment
  */
  "bar"
```

Given that, I think the current definition (where single-line comments are supported but multi-line aren't) is reasonable.

**Reply by zkat** [reply]
ugh I really don't like these warts at all. I definitely think your latter examples, while possibly ugly and pathological, should definitely be something that _works_? Maybe?

**Reply by lilyball** [reply]
I took another look at the spec, and actually my comment was wrong, multiline comments are supported where appropriate. The reason being that the `ws` token includes `multi-line-comment`, and escline has `ws*`. So my "but I can't write" example actually should parse, as should my final example. My penultimate example should be a parse error, which is correct as the backslash is not actually escaping anything.

**Reply by zkat** [reply]
ah ok! Yeah that seems fine.
